lib/matplotlib/dviread.py

```diff
@@ -862,38 +902,43 @@ def _register(self, words):
         # http://article.gmane.org/gmane.comp.tex.pdftex/4914
 
         texname, psname = words[:2]
-        effects, encoding, filename = '', None, None
-        for word in words[2:]:
-            if not word.startswith('<'):
-                effects = word
-            else:
-                word = word.lstrip('<')
-                if word.startswith('[') or word.endswith('.enc'):
-                    if encoding is not None:
-                        matplotlib.verbose.report(
-                            'Multiple encodings for %s = %s'
-                            % (texname, psname), 'debug')
-                    if word.startswith('['):
-                        encoding = word[1:]
-                    else:
-                        encoding = word
-                else:
-                    assert filename is None
-                    filename = word
-
-        eff = effects.split()
-        effects = {}
-        try:
-            effects['slant'] = float(eff[eff.index('SlantFont')-1])
-        except ValueError:
-            pass
-        try:
-            effects['extend'] = float(eff[eff.index('ExtendFont')-1])
-        except ValueError:
-            pass
+        words = words[2:]
+        effects, encoding, filename = b'', None, None
+
+        # pick the last non-filename word for effects
+        effects_words = [word for word in words if not word.startswith(b'<')]
+        if effects_words:
+            effects = effects_words[-1]
+
+        encoding_re = br'<<?(\[.*|.*\.enc)'
+        encoding_files = [word.lstrip(b'<').lstrip(b'[')
+                          for word in words
+                          if re.match(encoding_re, word)]
+        if len(encoding_files) > 1:
+            matplotlib.verbose.report(
+                'Multiple encodings for %s = %s' % (texname, psname), 'debug')
+        if encoding_files:
+            encoding = encoding_files[-1]
+
+        font_files = [word.lstrip(b'<')
+                      for word in words
+                      if word.startswith(b'<')
+                      and not re.match(encoding_re, word)]
```

Comment:
Not sure that 3 list comprehensions over `words` was the best change, efficiency-wise.